### ISSUE TITLE
Index without 0 in xerr/yerr causes KeyError

### ISSUE THREAD

**Issue by diazona** [pydata/pandas#11858]
When creating an errorbar plot, if the data object being used for the errors (`xerr` or `yerr`) doesn't have 0 in its index, it produces a `KeyError` from matplotlib code. I can produce this with the following code sample:

```python
import pandas as pd, numpy as np
i = np.array([1,2,3])
a = pd.DataFrame(i, index=i)
a.plot(yerr=a)
```

The underlying bug (if it is a bug) in matplotlib is responsible for several other issues, including #4493 and #6127, but this case is different because it uses only Pandas API methods, rather than passing a Pandas object to a matplotlib method. So it's a little harder to justify passing this off on matplotlib to fix, as was done in e.g. #6127.

If the "proper" fix ever is implemented in matplotlib code, it should solve this as well as #4493 and #6127 and all the others of that nature, but until that point, Pandas can work around it by converting the error object (if it is a `Series` or `DataFrame`) to an `ndarray`. I'm working on this in 4b04f80c41684ba9ab05ce8c87b17b96fde87290 but I'm not sure if there's a better way to fix it, or if this breaks something. (If so, the tests don't indicate it.)

**Comment by jorisvandenbossche** [comment]
@diazona This also seems a regression (in pandas or matplotlib), as I don't get an error with pandas 0.16.2 and matplotlib 1.4.3, but do get one with 0.17.1 and 1.5.0

**Comment by TomAugspurger** [comment]
Haven't looked, but sounds similar to https://github.com/matplotlib/matplotlib/issues/5550

@diazona I'm assuming you're using matplotlib 1.5? Can you try with 1.4.3?

**Comment by diazona** [comment]
Yeah, matplotlib#5550 seems like the exact same issue. That didn't come up in my search, so I hadn't realized the matplotlib developers were open to fixing it. If they do implement that fix, it renders my patch obsolete.

Let me take some time to get a previous version of matplotlib and test with that. I am indeed using 1.5.0. Here's the complete version info:

    >>> pd.show_versions()

    INSTALLED VERSIONS
    ------------------
    commit: 4b04f80c41684ba9ab05ce8c87b17b96fde87290
    python: 3.4.3.final.0
    python-bits: 64
    OS: Darwin
    OS-release: 15.2.0
    machine: x86_64
    processor: i386
    byteorder: little
    LC_ALL: None
    LANG: None

    pandas: 0.17.1+84.g4b04f80
    nose: 1.3.7
    pip: 7.1.2
    setuptools: 18.5
    Cython: 0.23.4
    numpy: 1.10.1
    scipy: 0.16.1
    statsmodels: 0.6.1
    IPython: 4.0.1
    sphinx: None
    patsy: 0.4.1
    dateutil: 2.4.2
    pytz: 2015.7
    blosc: None
    bottleneck: 1.0.0
    tables: 3.2.2
    numexpr: 2.4.6
    matplotlib: 1.5.0
    openpyxl: None
    xlrd: None
    xlwt: None
    xlsxwriter: None
    lxml: None
    bs4: None
    html5lib: None
    httplib2: None
    apiclient: None
    sqlalchemy: None
    pymysql: None
    psycopg2: None
    jinja2: 2.8

**Comment by diazona** [comment]
Update: matplotlib 1.4.3 indeed does not exhibit the error. After an enthralling `git bisect` run, I traced the `KeyError` to code introduced in 5f33e7da3c466fe0c74e1e7087548ec941bb228b, on the 1.5.0 development branch.

For what it's worth, the current master HEAD 9f6bb901f92a281dac54dde30b3c4319cd727d3f _does_ raise the `KeyError`.

I haven't tested with other versions of Pandas.

**Comment by tacaswell** [comment]
Can someone re-check if this is still a problem?  I think we fixed this is v1.5.1.

**Comment by andres-aguilar** [comment]
yes, here v. 1.5.1 and the same problem
